pkg/loadbalancer/server.go

```diff
@@ -113,9 +115,26 @@ func (s *Server) EnsureLoadBalancerDeleted(ctx context.Context, clusterName stri
 	return container.Delete(loadBalancerName(clusterName, service))
 }
 
-// loadbalancer name = cluster-name + service.namespace + service.name
+// loadbalancer name is a unique name for the loadbalancer container
 func loadBalancerName(clusterName string, service *v1.Service) string {
-	return constants.ContainerPrefix + "-" + clusterName + "-" + service.Namespace + "-" + service.Name
+	const nameMaxLen = 63
+	name := loadBalancerSimpleName(clusterName, service)
+
+	// create a hash of the name that will be 40 characters long
+	h := sha1.New()
+	io.WriteString(h, name)
+	name = constants.ContainerPrefix + "-" + fmt.Sprintf("%x", h.Sum(nil))
+
+	// This should never happen but just in case
+	if len(name) >= nameMaxLen {
+		return name[:nameMaxLen]
+	}
+
+	return name
```

Comment:
Let's use the same as kube-proxy
```suggestion
	hash := sha256.Sum256([]byte(loadBalancerSimpleName(clusterName, service)))
	encoded := base32.StdEncoding.EncodeToString(hash[:])
	name := constants.ContainerPrefix + "-" + encoded[:40]

	return name
```